alacritty/src/display.rs

```diff
@@ -498,8 +517,29 @@ impl Display {
             });
         }
 
+        #[cfg(not(any(target_os = "macos", windows)))]
+        self.request_frame(&self.window);
+
         self.window.swap_buffers();
     }
+
+    /// Requst a new frame for a window on Wayland.
+    #[inline]
+    #[cfg(not(any(target_os = "macos", windows)))]
+    fn request_frame(&self, window: &Window) {
+        // Register wayland frame callback
+        if let Some(surface) = window.wayland_surface() {
+            let proxy: Proxy<WlSurface> = unsafe { Proxy::from_c_ptr(surface as _) };
+            let attached = proxy.attach(self.wayland_event_queue.as_ref().unwrap().token());
```

Comment:
The walyand event queue is obviously always present when the wayland surface is, but it doesn't seem ideal to handle things this way.

Do you know if it's possible for the wayland surface to ever change? Would it be an option to always keep the event queue attached for example and store that on Display?

Reply:
It shouldn't change, I could try to store it on a window/display.

Reply:
Might be worth a shot? Not sure if it'll clean things up.